rust/agama-lib/src/storage/client.rs

```diff
@@ -379,4 +271,159 @@ impl<'a> StorageClient<'a> {
             Ok(None)
         }
     }
+
+    async fn build_component(
+        &self,
+        object: &DbusObject,
+    ) -> Result<Option<Component>, ServiceError> {
+        let properties = self.get_interface(object, "org.opensuse.Agama.Storage1.Component");
+
+        if let Some(properties) = properties {
+            Ok(Some(Component {
+                component_type: get_property(properties, "Type")?,
+                device_names: get_property(properties, "DeviceNames")?,
+                devices: get_property(properties, "Devices")?,
+            }))
+        } else {
+            Ok(None)
+        }
+    }
+
+    async fn build_drive(&self, object: &DbusObject) -> Result<Option<Drive>, ServiceError> {
+        let properties = self.get_interface(object, "org.opensuse.Agama.Storage1.Drive");
+
+        if let Some(properties) = properties {
```

Comment:
np: This is perfectly fine, but I would have used a [let-else](https://doc.rust-lang.org/beta/rust-by-example/flow_control/let_else.html) construction. Nothing to change today, anyway.